### ISSUE TITLE
Kamel cli should send exit code when integration in --dev mode is deleted

### ISSUE THREAD

**Issue by bfitzpat** [apache/camel-k#938]
In the tooling I'm having some difficulty finding ways to kill a --dev mode integration effectively. 

Essentially we're using Node.js child_process calls to spawn a cli call (i.e. kamel run "myintegration.groovy" --dev) and then we're listening for console output and bouncing that to a VS Code output channel. The problem is we don't have a clear signal to say that the --dev mode session is finished. We can't exactly do a Ctrl+C to stop the kamel --dev session from inside the tooling (though I am trying to find a way to replicate that). 

It looks like we're getting an exit signal on an error - https://github.com/apache/camel-k/blob/97d8d726d5543f501afb6bd8901345b3127f09ad/cmd/kamel/main.go#L47 so it seems we are sending a code on exit in some places (like when an integration stops - https://github.com/apache/camel-k/blob/97d8d726d5543f501afb6bd8901345b3127f09ad/pkg/cmd/run.go#L195)

Because of this, we're running a kamel --dev integration and then removing it with kamel delete. But that's not ever killing our initial kamel session - we're not getting a close or exit signal that says that it's done. 

Is there any chance that the kamel cli can send that exit/close signal when a --dev integration is deleted? I know it's a bit of an edge case, but it would help us out in the tooling.

**Comment by bfitzpat** [comment]
I have tried sending a kill signal to the running process and neither SIGINT nor SIGKILL seems to do any good. It is like one of the undead, hanging on well beyond its expiration date. So I'm not sure how to replicate the Ctrl+C when killing an integration running in the console in --dev mode.

If anybody has any ideas, I'm open!

**Comment by bfitzpat** [comment]
I eventually found a way round it. It turned out that the node.js child process "exec" call was creating a process and a sub process every time I invoked a --dev integration, so I could kill one process but the subprocess was just hanging around like the undead. I've moved to using the child process "spawn" call and now it behaves better. I can kill the process and it's truly dead. 

Would still like to get some kind of relationship between the "delete" call in the cli and a running --dev mode integration so they know about one another. But at least in the tooling I have a way to simulate that good old "ctrl+c" in the console and it works. :)
